web/src/localstorage.ts

```diff
@@ -155,18 +189,26 @@ export const localstorage = function () {
         },
 
         // remove a key with a given version.
-        remove(name) {
+        remove(name: string): void {
             ls.removeData(_data.VERSION, name);
         },
 
         // Remove keys which (1) map to a value that satisfies a
         // property tested by `condition_checker` AND (2) which
         // match the pattern given by `name`.
-        removeDataRegexWithCondition(name, condition_checker) {
+        removeDataRegexWithCondition(
+            name: string,
+            condition_checker: (value: string | null | undefined) => boolean,
+        ) {
             ls.removeDataRegexWithCondition(_data.VERSION, name, condition_checker);
         },
 
-        migrate(name, v1, v2, callback) {
+        migrate(
+            name: string,
+            v1: number,
+            v2: number,
+            callback: (data: object) => object,
+        ): object | undefined {
```

Comment:
`object` -> `unknown`

Reply:
Should `unknwon | undefined` be together, or unknown include the cases of undefined?